src/tensorlake/builder/client_v2.py

```diff
@@ -0,0 +1,251 @@
+        """
+        Create an instance of the ImageBuilderV2Client using environment variables.
+
+        The API key is retrieved from the TENSORLAKE_API_KEY environment variable.
+        The build service URL is retrieved from the INDEXIFY_URL environment variable,
+        defaulting to "https://api.tensorlake.ai" if not set.
+
+        The TENSORLAKE_BUILD_SERVICE environment variable can be used to specify
+        a different build service URL, mainly for debugging or local testing.
+
+        Returns:
+            ImageBuilderV2Client: An instance of the ImageBuilderV2Client.
+        """
+        api_key = os.getenv("TENSORLAKE_API_KEY")
+        server_url = os.getenv("INDEXIFY_URL", "https://api.tensorlake.ai")
+        build_url = os.getenv("TENSORLAKE_BUILD_SERVICE", f"{server_url}/images/v2")
+        return cls(build_url, api_key)
+
+    async def build_collection(
+        self, context_collection: Dict[Image, BuildContext]
+    ) -> Dict[str, str]:
+        """
+        Build a collection of images using the provided build context.
+
+        Args:
+            context_collection (Dict[Image, BuildContext]): A dictionary mapping images to their build contexts.
+        Returns:
+            dict: A dictionary mapping image hashes to their corresponding build IDs.
+        """
+        click.echo("Building images...")
+
+        builds = {}
+        for image, context in context_collection.items():
+            click.echo(f"Building {image.image_name}")
+            build = await self.build(context, image)
+            click.echo(f"Built {image.image_name} with hash {image.hash()}")
+            builds[image.hash()] = build.id
+
+        return builds
+
+    async def build(self, context: BuildContext, image: Image) -> BuildInfo:
+        """
+        Build an image using the provided build context.
+
+        Args:
+            context (BuildContext): The build context containing information about the graph,
+                                    graph version, and function name.
+            image (Image): The image to be built.
+        Returns:
+            dict: The response from the image builder service.
+        """
+        click.echo(
+            f"Building {context.graph_name} version {context.graph_version} for {context.function_name}"
+        )
+
+        _fd, context_file = tempfile.mkstemp()
+        image.build_context(context_file)
+
+        click.echo(
+            f"{context.graph_name}: Posting {os.path.getsize(context_file)} bytes of context to build service...."
+        )
+
+        files = {}
+        async with aiofiles.open(context_file, "rb") as fp:
+            files["context"] = await fp.read()
+
+        os.remove(context_file)
+        data = {
+            "graph_name": context.graph_name,
+            "graph_version": context.graph_version,
+            "graph_function_name": context.function_name,
+            "image_hash": image.hash(),
+        }
+
+        res = await self._client.put(
```

Comment:
Would adding retries help here for better ux in case e.g. if we deploy image builder service and it's temporarily down?